### PR TITLE
Use unique name for AlertNotify tempfile

### LINKED ISSUE #6524: make check leaves behind /tmp/alertnotify.txt
Error arises when another user on same system runs a "make check" or otherwise tries to access /tmp/alertnotify.txt before /tmp is flushed by system and creates a permission clash. E.g.
Making check in src
make[1]: Entering directory '/home/user2/bitcoin-dev/bitcoin/src'
make[2]: Entering directory '/home/user2/bitcoin-dev/bitcoin/src'
make  check-TESTS check-local
make[3]: Entering directory '/home/user2/bitcoin-dev/bitcoin/src'
make[4]: Entering directory '/home/user2/bitcoin-dev/bitcoin/src'
../build-aux/test-driver: line 107:  5975 Aborted                 (core dumped) "$@" > $log_file 2>&1
FAIL: test/test_bitcoin
.
.
See src/test-suite.log
Please report to https://github.com/bitcoin/bitcoin/issues
.
.

 and in src/test-suite.log

Running 161 test cases...
unknown location(0): fatal error in "AlertNotify": std::runtime_error: boost::filesystem::remove: Operation not permitted: "/tmp/alertnotify.txt"
test/alert_tests.cpp(154): last checkpoint
unknown location(0): fatal error in "PartitionAlert": memory access violation at address: 0x00000078: no mapping at fault address
test/alert_tests.cpp(154): last checkpoint

### PR DESCRIPTION
**Pull request by casey**
Fixes #6524

Use a unique name for alertnotify.txt tempfile during AlertNotify test

### CODE REVIEW

File: src/test/alert_tests.cpp [hunk 1]
```diff
@@ -162,8 +162,8 @@ BOOST_AUTO_TEST_CASE(AlertNotify)
     SetMockTime(11);
     const std::vector<unsigned char>& alertKey = Params(CBaseChainParams::MAIN).AlertKey();
 
-    boost::filesystem::path temp = GetTempPath() / "alertnotify.txt";
-    boost::filesystem::remove(temp);
+    boost::filesystem::path temp = GetTempPath() /
+        boost::filesystem::unique_path("alertnotify-%%%%.txt");
```
Comment (jonasschnelli) [inline src/test/alert_tests.cpp]: minor nit: wrong indent (same on L171 / not related to this PR)

**Comment by randy-waterhouse** [comment]
tested ACK

**Comment by jonasschnelli** [comment]
ACK.
Couldn't find any details about when files generated with boost::filesystem::unique_path are deleted.
I think adding a `boost::filesystem::remove(temp);` wouldn't harm?

**Comment by laanwj** [comment]
ACK. Using a hardcoded/predictable temporary file name can be dangerous.

And agree with @jonasschnelli that the test should try to clean up after itself.

File: src/test/alert_tests.cpp (hunk 1 above)
Comment (casey) [inline src/test/alert_tests.cpp]: Isn't the right indentation four spaces? Or is it different for a line continuation?

**Comment by casey** [comment]
@jonasschnelli @laanwj It actually does clean up after itself; there's a call to `boost::filesystem::remove(temp)` at the end of the function.

File: src/test/alert_tests.cpp (hunk 1 above)
Comment (jonasschnelli) [inline src/test/alert_tests.cpp]: Meh. Sorry. I oversaw the line break... all good!
You can always use `clang-format -i <file>` if you wan't to confirm to the code-style-guidelines.

**Comment by jonasschnelli** [comment]
@casey: Ah. Now i see how this works. The `boost::filesystem::remove(temp);` removed in this PR was there because the file was not really a random tmp file.

ACK.

**Comment by laanwj** [comment]
Oh! oops. Apologies for the spurious comment, then.

**Pull request merged**